### PR TITLE
CC-2158: make lvm monitoring configurable during serviced startup

### PR DESCRIPTION
**Pull request by smousa**

### CODE REVIEW

File: volume/devicemapper/devicemapper.go [hunk 1]
```diff
@@ -494,7 +497,25 @@ func (d *DeviceMapperDriver) ensureInitialized() error {
 		return err
 	}
 	if d.DeviceSet == nil {
-		deviceSet, err := devmapper.NewDeviceSet(poolPath, true, d.options, nil, nil)
+		var (
+			thinPoolDev         string
+			enableLVMMonitoring string
```
Comment (tgjonestx) [inline volume/devicemapper/devicemapper.go]: Shouldn't the default for this string be `"n"`?  We don't want customers to make a configuration change to opt-in to the no-monitoring feature.

File: volume/devicemapper/devicemapper.go [hunk 2]
```diff
@@ -494,7 +497,25 @@ func (d *DeviceMapperDriver) ensureInitialized() error {
...
+			if strings.HasPrefix(option, "dm.") {
+				dmoptions = append(dmoptions, option)
+				if strings.HasPrefix(option, "dm.thinpooldev=") {
+					thinPoolDev = strings.TrimPrefix(option, "dm.thinpooldev=")
+				}
+			} else if strings.HasPrefix(option, "enablelvmmonitoring=") {
+				enableLVMMonitoring = strings.TrimPrefix(option, "enablelvmmonitoring=")
+			} else {
+				glog.Errorf("Unable to parse option %s", option)
+				return ErrInvalidOption
```
Comment (iancmcc) [inline volume/devicemapper/devicemapper.go]: Maybe don't choke, just log it and move on.

File: volume/devicemapper/devicemapper.go [hunk 3]
```diff
@@ -504,6 +525,36 @@ func (d *DeviceMapperDriver) ensureInitialized() error {
...
+				cmd = exec.Command("lvchange", "--monitor", "y", thinPoolDev)
+				output, err := cmd.CombinedOutput()
+				if err != nil {
+					glog.Errorf("Could not run command %v: %s (%s)", cmd, string(output), err)
+					return err
+				}
+			} else {
+				glog.Warningf("Ignoring option 'enablelvmmonitoring'; no thin pool device specified")
+			}
+		case "n", "":
```
Comment (iancmcc) [inline volume/devicemapper/devicemapper.go]: Can we just use `default:` here?

File: volume/devicemapper/devicemapper.go (hunk 1 above)
Comment (tgjonestx) [inline volume/devicemapper/devicemapper.go]: NVM - I missed the extra `""` in the case statement below

File: volume/devicemapper/devicemapper.go (hunk 2 above)
Comment (smousa) [inline volume/devicemapper/devicemapper.go]: @jplouis prefers that it chokes in case someone fat-fingers something

**Pull request merged**